_posts/2017-09-05-Browser Security Features.md

```diff
@@ -0,0 +1,290 @@
+
+### Should you activate this on your site? 
+Absolutely! 
+Modern browsers are shifting from notifying users that a page is secure to warning them that it isn't.
+On top of that, Google gives a slight ranking boost to HTTPS sites.
+
+## HTTP Strict Transport Security
+Once your server is properly configured to use TLS, your next step is to redirect your users to the secure version.
+You could do this by simply adding a redirect-rule in your web server for the non-secure pages.
+This means that users will still first connect to your non-secure site, allowing a potential attacker to intercept the request and do his nefarious deeds.
+Wouldn't it be nice if you could tell the browser to just go straight to the secure version?
+That's the thinking behind the HSTS (HTTP Strict Transport Security) header. 
+
+HSTS simply tells the browser that you're expecting it to use HTTPS for a certain time.
+As a result, the browser will *automatically* replace http:// with https:// **before** making the call.
+This means that even following a link that explicitly defines http:// will instead be called using a secure connection.
+
+The configuration of HSTS is as easy as can be: you simply add the following header to your response:
+
+```Strict-Transport-Security: max-age=31536000; includeSubDomains```
+
+This will tell the browser that for the next 365 days, it should connect to your domain using https.
+The `includesubdomains` directive tells the browser that your subdomains should also be called using https.
+Setting the max-age to 0 tells the browser that you no longer wish your domain to be HSTS-enabled. 
+
+### HSTS preload
+Of course in this scenario, the user's first connection will still take place over an unsecured connection.
+This would offer an attacker a brief period in which he can still hijack the connection.
+To prevent this, most major browsers (Chrome, Firefox, Safari, Edge, IE11 and Opera) offer an HSTS preload list.
+Domains on this list will automatically be loaded over HTTPS from the start, without having to go through the HTTP -> HTTPS redirect.
+If you want your domain to be included in this list, you should add the `preload` directive to the HSTS header.
+
+```Strict-Transport-Security: max-age=31536000; includeSubDomains; preload```
+
+Afterwards, you can register your domain at [https://hstspreload.org/](https://hstspreload.org/){: target="_blank" }
+
+### Are there any risks?
+Activating HSTS does offer some risks:
+* If you include the `includesubdomains` directive, you tell the browser that *all* subdomains need to be retrieved over HTTPS.
+  If your internal applications are on a subdomain (e.g. internal.example.com), you'll block access to those that haven't enabled TLS yet.
+* Adding the `preload` directive is even more dangerous: because this tells browser makes to hardcode your HSTS settings.
```

Comment:
This sentence looks a bit off? "because this tells browser makes to hardcode your HSTS settings."
I'd also leave out the colon as it's being followed by a "because".